Improve documentation and revise naming

matthiasgeihs opened the pull request:
- Add missing documentation to `class Point`.
- Revise core kit interface documentation to include more details about function behavior and return values.
- Revise naming of internal functions related to backing up metadata share.

himanshuchawla009 commented on src/mpcCoreKit.ts:
```diff
@@ -461,10 +461,10 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
       await this.addFactorDescription(factorKey, shareDescription, additionalMetadata);
       if (!this.options.manualSync) await this.tKey.syncLocalMetadataTransitions();
-      return factorKey.toString("hex").padStart(64, "0");
+      return factorKey.toArrayLike(Buffer, "be", SCALAR_LEN).toString("hex");
```
we should use ``.toString("hex", 64);```

matthiasgeihs replied:
I agree this can be shortened using `factorKey.toString(...)`
However, we shouldn't use magic numbers like `64`. Instead we should use constants, or even better common classes or functions for serializing elliptic curve scalars and points.
Already added a class for `Point` for points before.
In this case, added a utility function `scalarBNToBufferSEC1` and using this here. It outputs a buffer of the correct length which can then be easily converted into different representations (e.g., "hex" or "base64"). This will hopefully help us avoid these discussions in the future.

himanshuchawla009 commented on src/mpcCoreKit.ts:
```diff
@@ -476,15 +476,14 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
 
     await deleteFactorAndRefresh(this.tKey, factorPub, this.state.factorKey, this.signatures);
-    const factorPubHex = Point.fromTkeyPoint(factorPub).toBufferSEC1(true).toString("hex");
+    const factorPubHex = fpp.toBufferSEC1(true).toString("hex");
```
toString("hex", 64)

ieow replied:
the convert to public key which is 64 bytes for raw , 65 bytes for full, and 33 bytes for compress
converted to hex would be 128, 130, and 66 chars

matthiasgeihs replied:
@himanshuchawla009 
`toBufferSEC1(...)` returns a `Buffer`. Its `toString` method does not what you think here. If you call `toString("hex", n)` on a `Buffer`, it will convert the buffer to hex starting from the `n`-th byte.

Moreover, we should avoid using magic numbers like in `toString("hex", 64)`, as pointed out in my comment above.

yashovardhan commented on src/interfaces.ts:
```diff
@@ -191,10 +195,12 @@ export interface ICoreKit {
+   * @returns The factor key.
    */
-  createFactor(CreateFactorParams: CreateFactorParams): Promise<string>;
+  createFactor(params: CreateFactorParams): Promise<string>;
```
for consistency across all functions, it should be `createFactorParams` not `param`

The `C` should be in lowercase.

matthiasgeihs replied:
My thinking was: Variable names should be concise. It's clear from context that these are parameters to the function `createFactor`, hence the name `createFactorParams` is redundant.

However, not a big deal. I can change it back for consistency, if you like. Needs to be lower case though.

matthiasgeihs replied:
changed as requested

yashovardhan commented on src/mpcCoreKit.ts:
```diff
@@ -443,10 +443,10 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
   }
 
-  public async createFactor(createFactorParams: CreateFactorParams): Promise<string> {
+  public async createFactor(params: CreateFactorParams): Promise<string> {
```
If you're looking to replace all with just `params`, in that case please update `enableMFAParams` and all others that are using similar naming convention. Either way works for us, but it should be consistent, no matter what we use.

matthiasgeihs replied:
see reply above

yashovardhan reviewed (commented):
Please rebase the PR to the latest as well before moving forward, left few minor comments, rest the renaming makes sense.

matthiasgeihs commented on src/mpcCoreKit.ts:
```diff
@@ -467,7 +467,10 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
+      return factorKey
+        .toArrayLike(Buffer, "be", SCALAR_LEN)
+        .toString("hex")
+        .padStart(SCALAR_LEN * 2, "0");
```
This padding is redundant. `toArrayLike(Buffer, "be", SCALAR_LEN)` already guarantees that it has the correct length.

ieow replied:
@matthiasgeihs , you are right

matthiasgeihs commented:
Addressed comments.
Started rebasing but found some issues with base branch:
https://github.com/Web3Auth/mpc-core-kit/pull/53
How should we deal with them? Should I fix them here. Need input before rebasing this one here. @ieow

ieow reviewed (approved):
lgtm
